quickstart/backend/src/main/java/com/digitalasset/quickstart/service/LicenseRenewalRequestsApiImpl.java

```diff
@@ -56,139 +56,146 @@ public LicenseRenewalRequestsApiImpl(
+                                                                return CompletableFuture.completedFuture(
+                                                                        ResponseEntity.status(HttpStatus.NOT_FOUND).build());
+                                                            }
+
+                                                            ContractId<License> licenseCid = maybeLicense.get().contractId;
+
+                                                            // 3. Fetch necessary disclosed contracts
+                                                            CompletableFuture<CommandsOuterClass.DisclosedContract> amuletRulesFut =
+                                                                    fetchAmuletRulesDisclosedContract();
+                                                            CompletableFuture<CommandsOuterClass.DisclosedContract> openMiningRoundFut =
+                                                                    fetchOpenMiningRoundDisclosedContract(miningRound);
+
+                                                            return CompletableFuture.allOf(amuletRulesFut, openMiningRoundFut)
+                                                                    .thenCompose(v -> {
+                                                                        CommandsOuterClass.DisclosedContract amuletRulesDc = amuletRulesFut.join();
+                                                                        CommandsOuterClass.DisclosedContract openMiningRoundDc = openMiningRoundFut.join();
+
+                                                                        AppTransferContext transferContext = new AppTransferContext(
+                                                                                new ContractId<>(amuletRulesDc.getContractId()),
+                                                                                new ContractId<>(openMiningRoundDc.getContractId()),
+                                                                                Optional.empty()
+                                                                        );
+
+                                                                        // 4. Construct the CompleteRenewal choice
+                                                                        LicenseRenewalRequest.LicenseRenewalRequest_CompleteRenewal choice =
+                                                                                new LicenseRenewalRequest.LicenseRenewalRequest_CompleteRenewal(
+                                                                                        acceptedPaymentCid,
+                                                                                        licenseCid,
+                                                                                        transferContext
+                                                                                );
+
```

Comment:
You have 5 steps here, collecting and collating business values required for the operation, and no logs.

You need an INFO log here with the key business values, the commandId, the contractId, and probably 1 or 2 supporting contractIds as well.

You might also consider if any of the prior steps are worth a DEBUG entry.